libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayUpdate.kt

```diff
@@ -0,0 +1,5 @@
+package com.mapbox.navigation.core.replay.history
+
+data class ReplayUpdate(
+    val replayEventHappened: List<ReplayEventBase>
```

Comment:
Why is this a list? Is this because multiple events can happened at the same interval (1 second per `replayUpdateSpeedMillis`)?

Reply:
Here is a bubble diagram of the different forms of time. using a list, we can create our own simulator loop window that create buckets of events. this first version is attempting to be simple, so it'll replay the events in bunches. a future perfect simulator would replay each window accurately.

history event time (each number is a unique event)
12----3---45---6--7----8-- 

replay event time (replayUpdateSpeedMillis in code)
-----o---------o---------o

this version of simulator
-----12--------345--------678

future perfect simulator
-----12----3---45---6--7----8-- 

... another approach is to have a really fast replay event time, but it'd still need a list because some events happen at the same time

Reply:
> future perfect simulator
> -----12----3---45---6--7----8--

To be perfect should be like this 👇 right? :trollface: 

```12----3---45---6--7----8-------```

Reply:
haha yeah.. but i don't have a good solution for _that perfect_. maybe can look into [value prediction](https://www.microarch.org/cvp1/) ideas 🙃